File: Counting-Frequencies-of-Array-Elements.js
class Solution {
    countFrequencies(nums) {
        const visited = new Set();
        const result = [];

        for (let i = 0; i < nums.length; i++) {
            if (visited.has(nums[i])) continue;

            let count = 0;
            for (let j = 0; j < nums.length; j++) {
                if (nums[i] === nums[j]) {
                    count++;
                }
            }

            result.push([nums[i], count]);
            visited.add(nums[i]);
        }
        return result;
    }
}
const n = [1, 2, 2, 1, 3];
console.log(new Solution().countFrequencies(n));

File: Counting-Frequencies-of-Array-Elements.py
class Solution:
    def countFrequencies(self, nums):
        # Your code goes here
        visited=set()
        result=[]
        for i in range(len(nums)):
            if nums[i] in visited:
                continue
            count=0
            for j in range(len(nums)):
                if nums[i]==nums[j]:
                    count+=1
            result.append([nums[i],count])
            visited.add(nums[i])
        return result
raw = input().strip()       # "[1,2,2,1,3]"
raw = raw.strip("[]")       # "1,2,2,1,3"
n = list(map(int, raw.split(",")))
print(Solution().countFrequencies(n))

